modules/virt-cnv-virt-alerts.adoc

```diff
@@ -0,0 +1,487 @@
+
+.Troubleshoot
+
+Determine a virt-operator pod's leader status from the pod logs. The log messages containing `Started leading` and `acquire leader` indicate the leader status of a given virt-operator pod.
+
+Additionally, always check if there are any running virt-operator pods and the pods' statuses with these commands:
+[source,terminal]
+----
+export NAMESPACE="$(oc get kubevirt -A -o custom-columns="":.metadata.namespace)"
+
+oc -n $NAMESPACE get pods -l kubevirt.io=virt-operator
+
+oc -n $NAMESPACE logs <pod-name>
+
+oc -n $NAMESPACE describe pod <pod-name>
+----
+
+*Leader pod example:*
+[source,terminal]
+----
+$ oc -n $NAMESPACE logs <pod-name> |grep lead
+{"component":"virt-operator","level":"info","msg":"Attempting to acquire leader status","pos":"application.go:400","timestamp":"2021-11-30T12:15:18.635387Z"}
+I1130 12:15:18.635452       1 leaderelection.go:243] attempting to acquire leader lease <namespace>/virt-operator...
+I1130 12:15:19.216582       1 leaderelection.go:253] successfully acquired lease <namespace>/virt-operator
+{"component":"virt-operator","level":"info","msg":"Started leading","pos":"application.go:385","timestamp":"2021-11-30T12:15:19.216836Z"}
+----
+
+*Non-leader pod example:*
+[source,terminal]
+----
+$ oc -n $NAMESPACE logs <pod-name> |grep lead
+{"component":"virt-operator","level":"info","msg":"Attempting to acquire leader status","pos":"application.go:400","timestamp":"2021-11-30T12:15:20.533696Z"}
+I1130 12:15:20.533792       1 leaderelection.go:243] attempting to acquire leader lease <namespace>/virt-operator...
+----
+
+.Resolution
+
+There are several reasons for no virt-operator pod holding the leader lease, despite one or more virt-operator pods being in `Ready` state. Identify the root cause and take appropriate action.
+
+Otherwise, open a support issue and provide the information gathered in the troubleshooting process.
+
+//NoReadyVirtController Alert
+[id="NoReadyVirtController_{context}"]
+== NoReadyVirtController alert
+
+.Description
+
+The virt-controller monitors virtual machine instances (VMIs). The virt-controller also manages the associated pods by creating and managing the lifecycle of the pods associated with the VMI objects.
+
+A VMI object always associates with a pod during its lifetime. However, the pod instance can change over time because of VMI migration.
+
+This alert occurs when detection of no ready virt-controllers occurs for five minutes.
+
+.Reason
+
+If the virt-controller fails, then VM lifecycle management completely fails. Lifecycle management tasks include launching a new VM instance or shutting down an existing VM instance.
+
+.Troubleshoot
+
+. Modify the environment variable `NAMESPACE`.
++
+[source,terminal]
+----
+export NAMESPACE="$(oc get kubevirt -A -o custom-columns="":.metadata.namespace)"
+----
+
+. Run this command:
```

Comment:
It would be nice to have some explanation on what is to expect after running the command or why it is being run, specially since we are checking the deployment again at line 112.

Reply:
@iholder-redhat Can you please help here?

Reply:
I think here the Trouleshooting paragraph is mixed with Resolution. 

**Troubleshouting (gathering info):**

1. Check deployment status for available replicas and conditions:
`$ oc -n $NAMESPACE get deployment virt-controller -o yaml`

2. check if virt-controller pods exists and their status:
`oc -n $NAMESPACE get pod | grep virt-controller`

3. check virt-controller pods events:
`oc -n $NAMESPACE describe pod <virt-controller pod> `

4. check virt-controller pods logs:
`oc -n $NAMESPACE logs <virt-controller pod> `

5. . Check if there are issues with the nodes, such as if the nodes are in a `NotReady` state.



**Resolution:** - something like we have for another alerts:

There are several reasons for no virt-controller pods being in a `Ready` state. Identify the root cause and take appropriate action.

Otherwise, open a support issue and provide the information gathered in the troubleshooting process.